dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/runner/WorkerTaskExecuteRunnable.java

```diff
@@ -279,6 +284,25 @@ protected void sendTaskResult() {
                 taskExecutionContext.getCurrentExecutionStatus());
     }
 
+    protected void sendRemoteLogIfNeeded() {
+        if ("true".equalsIgnoreCase(PropertyUtils.getString(Constants.REMOTE_LOGGING_ENABLE))) {
+            try {
+                logger.info("Start to send log {} to remote target {}", taskExecutionContext.getLogPath(),
+                        PropertyUtils.getString(Constants.REMOTE_LOGGING_TARGET));
+                logger.info("Wait log {} to be flushed...", taskExecutionContext.getLogPath());
+                Thread.sleep(5000);
```

Comment:
why we have to sleep for 5s here? can we run continue when we get log path immediately?

Reply:
The reason for waiting here is to allow the final output of the log to be written to the local file
Here is an example.

<img width="1080" alt="截屏2023-01-05 10 19 26" src="https://user-images.githubusercontent.com/38122586/210687054-c0a787bf-75c8-4cd2-9a71-077d24464ea4.png">

The green box in the figure is the last output of the task log, which is later than the time when the log is uploaded to the remote storage.

Therefore, if the thread does not use `sleep()`, the log uploaded to the remote storage will lose the statements in the green box.

Reply:
If you trigger `sendRemoteLogIfNeeded` in `afterExecute`, it does have to sleep for a few seconds, because when handling large amount of logs, async flush will take a lot of time. 

So, I recommend you trigger in `logHandle` when handling `FINALIZE_SESSION_MARKER` which marks that log appender will be closed in few seconds, flush logic in ds(you can check in `parseProcessOutput`) makes sure that no logs will reach in these few seconds and we don't have to wait for five seconds any more.

Reply:
> If you trigger `sendRemoteLogIfNeeded` in `afterExecute`, it does have to sleep for a few seconds, because when handling large amount of logs, async flush will take a lot of time.
> 
> So, I recommend you trigger in `logHandle` when handling `FINALIZE_SESSION_MARKER` which marks that log appender will be closed in few seconds, flush logic in ds(you can check in `parseProcessOutput`) makes sure that no logs will reach in these few seconds and we don't have to wait for five seconds any more.

Hi, @Radeity Thanks a lot for your suggestion! I think this is a good idea and I'll look into it.